doc/safegcd_implementation.md

```diff
@@ -126,6 +126,27 @@ The approach here avoids the need for 2x2 matrix multiplications of various size
 be faster at the level of optimization we're able to do in C.
 
 
+## 2b. From GCDs to Jacobi symbol
+
+We can also use a similar approach to calculate Jacobi symbol (x | M) by keeping track of an extra variable j, for which at every step (x | M) = j * (g | f). As we update f and g, we make corresponding updates to j using [properties of the Jacobi symbol](https://en.wikipedia.org/wiki/Jacobi_symbol#Properties). In particular, we update j whenever we divide g by 2 or swap f and g; these updates depend only on the values of f and g modulo 4 or 8, and can thus be applied very quickly. Overall, this calculation is slightly simpler than the one for modular inverse because we no longer need to keep track of d, e.
```

Comment:
Can you use the same style as the rest of the document? That means putting mathematical expressions between `*` symbols to make it italic.

Reply:
Done!